shell-ui/src/initFederation/ConfigurationProviders.tsx

```diff
@@ -318,9 +341,52 @@ export function useDiscoveredViews(): ViewDefinition[] {
 
     return [];
   }) as ViewDefinition[];
-
   return discoveredViews;
 }
+
+type FederatedRoute = {
+  app: SolutionUI;
+  view: View;
+};
+export function useFederatedRoutes(): FederatedRoute[] {
+  const { retrieveConfiguration } = useConfigRetriever();
+  const { retrieveDeployedApps } = useDeployedAppsRetriever();
+  const deployedApps = retrieveDeployedApps();
+
+  deployedApps.forEach((app) => {
+    if (app.appHistoryBasePath.endsWith('/')) {
+      throw new Error(
+        `appHistoryBasePath of app ${app.name} ends with a /, this is not allowed`,
+      );
+    }
+  });
+
+  const federatedRoutes = deployedApps.flatMap((app) => {
+    const appBuildConfig = retrieveConfiguration<'build'>({
+      configType: 'build',
+      name: app.name,
+      url: app.url,
+    });
+
+    if (!appBuildConfig) {
+      return [];
+    }
```

Comment:
nit: we are iterating twice on deployedApps within the same scope, can we iterate only once on this collection?